pkg/cmd/root.go

```diff
@@ -61,6 +62,21 @@ var rootCmd = &cobra.Command{
 		// tty is required to accept user input, otherwise the update can't be accepted/declined
 		isTTY := isatty.IsTerminal(os.Stdout.Fd())
 
+		// version check
+		if !configuration.GetFlag(models.FlagUpdateCheck) {
+			version.PerformVersionCheck = false
+		}
+		if version.PerformVersionCheck && configuration.CanReadEnv {
+			enable := os.Getenv("DOPPLER_ENABLE_VERSION_CHECK")
+			if enable == "false" {
+				logValueFromEnvironmentNotice("DOPPLER_ENABLE_VERSION_CHECK")
+				version.PerformVersionCheck = false
+			}
+		}
+		if version.PerformVersionCheck {
+			version.PerformVersionCheck = !utils.GetBoolFlagIfChanged(cmd, "no-check-version", !version.PerformVersionCheck)
+		}
```

Comment:
Relocated this logic from `loadFlags()` since the config file isn't loaded yet when that function runs.